salt/states/pkg.py

```diff
@@ -872,6 +880,27 @@ def _nested_output(obj):
     ret = nested.output(obj).rstrip()
     return ret
 
+def _resolve_capabilities(pkgs, refresh=False, **kwargs):
+    '''
+    Resolve capabilities in ``pkgs`` and exchange them with real package
+    names, when the result is distinct.
+    This feature can be turned on while setting the paramter
+    ``resolve_capabilities`` to True.
+
+    Return the input dictionary with replaced capability names and as
+    second return value a bool which indicate if a refresh was run.
+
+    In case of ``resolve_capabilities`` is False (disabled) or not
+    supported by the implementation the input is returned unchanged.
+    '''
+    was_refreshed = False
+    if not pkgs or 'pkg.resolve_capabilities' not in __salt__:
```

Comment:
How exactly this can happen? There are nothing of decorators that would be able to remove this function from being lazy-loaded. Am I missing something?..

Reply:
We are in states/pkg and only zypper module implements pkg.resolve_capabilities.
If yum is loaded this function is not available.